### PR TITLE
Updates to plugin to support version .3 of the CSI spec

### PR DESCRIPTION
**Pull request by matthias50**
* Change CSI spec version to v0.3.0
* Tweak go code and regenerate vendor directory to fix compile errors

Change-Id: Id82184331deece289221bbea0718e933cac46d4f

### CODE REVIEW

**Comment by matthias50** [comment]
I've been doing some experiments trying to get this plugin to work with k8s. After updating to .3 it partially works, so I'd like to get this change in before moving on to other issues I'm seeing with k8s + csilvm.

**Comment by gpaul** [comment]
Thanks a lot for the contribution. 

CI failed due to some linting errors. Please run `make check`. 

We'll also have to rebase this PR on the latest master.

File: pkg/csilvm/validate.go [hunk 1]
```diff
@@ -146,8 +146,21 @@ func validateCapacityRange(capacityRange *csi.CapacityRange) error {
...
+
+	// limit_bytes of 0 is equivalent to not setting a limit, so this is
+	// allowed. We already know required_bytes is non-zero because of the above
+	// check
+	if capacityRange.GetLimitBytes() == 0 {
+		return nil
+	}
 	if capacityRange.GetRequiredBytes() > capacityRange.GetLimitBytes() {
-		return ErrCapacityRangeInvalidSize
+		// return ErrCapacityRangeInvalidSize
```
Comment (jdef) [inline pkg/csilvm/validate.go]: nit: remove dead code

File: Gopkg.toml [hunk 2]
```diff
@@ -27,19 +27,19 @@ required = [
 
 [[constraint]]
   name = "github.com/container-storage-interface/spec"
-  version = "v0.2.0"
+  version = "v0.3.0"
 
-[[constraint]]
+[[override]]
```
Comment (jdef) [inline Gopkg.toml]: unrelated to the goal of this PR: unclear why gogo is listed as a dependency here. we should remove this.

File: pkg/csilvm/validate.go (hunk 1 above)
Comment (matthias50) [reply pkg/csilvm/validate.go]: Done

File: Gopkg.toml (hunk 2 above)
Comment (gpaul) [reply Gopkg.toml]: This (should?) already be removed through rebase.

**Comment by gpaul** [comment]
Thanks @matthias50 - the change to CreateVolume's validation caused 1 test to fail:

https://github.com/mesosphere/csilvm/pull/94/commits/fceecd5419205cf26cb2e8f6f7060048f1fcd3b0#diff-9bc9954444314954aefd3d6c72ff9ed4R153

And the failing test is:
`TestCreateVolumeVolumeCapabilitiesCapacityRangeRequiredLessThanLimit`

As the name suggests, it checks that required is less than limit. I suspect that your change makes it hit a different validation error case. It used to expect `ErrCapacityRangeInvalidSize` (https://github.com/mesosphere/csilvm/pull/94/commits/45ca7d702bdbe734801883a6cf926923b7365ef7#diff-9bc9954444314954aefd3d6c72ff9ed4L157) and now returns a custom error.

You can fix that by creating exactly the expected error in the test:

```
expErr := status.Errorf(
			codes.InvalidArgument,
			"required_bytes: %d cannot exceed the limit_bytes: %d",
			capacityRange.GetRequiredBytes(),
			capacityRange.GetLimitBytes(),
)
```

And then checking for that:

```
if !grpcErrorEqual(err, expErr) {
		t.Fatal(err)
}
```

Please run:

go test -c -i ./pkg/csilvm && sudo unshare -m -- ./csilvm.test -test.v -test.run=TestCreateVolumeVolumeCapabilitiesCapacityRangeRequiredLessThanLimit

**Comment by matthias50** [comment]
@gpaul can you merge this for me?

**Comment by gpaul** [comment]
Done

**Pull request merged**